File: 백준/Silver/11650. 좌표 정렬하기/좌표 정렬하기.java
import java.util.*;
import java.io.*;


public class Main {
	

	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int N = Integer.parseInt(br.readLine());
		List<int[]> list = new ArrayList<>();
		StringTokenizer st;
		for(int i = 0; i < N; i++) {
			st = new StringTokenizer(br.readLine());
			int x = Integer.parseInt(st.nextToken());
			int y = Integer.parseInt(st.nextToken());
			list.add(new int[] {x, y});
		}
		
		list.sort(Comparator.comparingInt((int[] a)-> a[0])
				.thenComparingInt((int[] a )-> a[1]));
		
		
		StringBuilder sb = new StringBuilder();
		for(int[] point :list) {
			sb.append(point[0]).append(" ").append(point[1]).append("\n");
		}

		System.out.println(sb);
	}

}

File: 백준/Silver/11650. 좌표 정렬하기/좌표 정렬하기.kt
package org.example

import java.util.StringTokenizer

fun main() {
    val br = System.`in`.bufferedReader()
    val bw = System.`out`.bufferedWriter()
    val sb = StringBuilder()
    val N = br.readLine().toInt()
    val points = mutableListOf<Pair<Int, Int>>()
    repeat(N) {
        val st = StringTokenizer(br.readLine())
        points.add(st.nextToken().toInt() to st.nextToken().toInt())
    }
    points.sortedWith(compareBy<Pair<Int, Int>> { it.first }
        .thenBy { it.second })
        .forEach {
            sb.append("${it.first} ${it.second}\n")
        }
    bw.write(sb.toString())
    bw.flush()
    bw.close()
    br.close()
}